fix: Change tests to use new filter and use gax warn for warning just once

danieljbruce opened the pull request:
**Summary:**
To partially address https://github.com/googleapis/nodejs-datastore/issues/1109 we use a warning function internally that will emit a warning only when the legacy `.filter` function is used the first time. Right now, a warning is emitted every time the legacy filter function is used.

**Changes:**
**In src/query.ts:**
Changes are made inside the src/query.ts file to use the function that will limit warnings to 1.
**In tests outside of test/query.ts:**
.filter is called the new way by providing PropertyFilter and CompositeFilter objects, this ensures that when the test/query.ts file is run, a warning about using .filter the legacy way hasn't been emitted yet
**In test/query.ts:**
A test is moved up so that it is the first test that uses the legacy filter function. This test expects a warning so this change is necessary to make this test pass.

danieljbruce committed:
trim operator

In the legacy version of filter we trim the operator. We should do this here too.

danieljbruce committed:
In test cases, replace various filter calls

Always pass entity filters into the filter function in tests to reflect new behavior.

danieljbruce committed:
accept string too

A string is needed as an input parameter because we might trim it to get an operator.

danieljbruce committed:
Use .filter the old way

Revert changes on the .filter method

danieljbruce committed:
Change docs to use new filter

The new property filter should be used and this should be reflected in the doc comments.

danieljbruce committed:
Use new Property filter

In entity.ts, use the new property filter constructor.

danieljbruce committed:
Revert changes to operator

The changes cause compiler errors. Do not trim the operator for now.

danieljbruce committed:
Move test up and add change

Add change to issue warning just once. Use google-gax so that the warning is issued just once.

danieljbruce committed:
Revert "Change docs to use new filter"

This reverts commit c290481f9442f46cb6ddc84f15914342d7435b8a.

danieljbruce committed:
run linter

feywind commented on test/query.ts:
```diff
@@ -206,6 +206,18 @@ describe('Query', () => {
   });
 
   describe('filter', () => {
+    it('should issue a warning when a Filter instance is not provided', done => {
```
It might be nice to make this validate that the warning only happens once, but since gax probably also tests that, it's probably not a big issue.

danieljbruce replied:
Done

danieljbruce committed:
Add a test to ensure just one warning

A test to make sure the warning is emitted just once might provide proper coverage.
